src/test/java/gift/ControllerTest/KakaoOAuthControllerTest.java

```diff
@@ -32,32 +31,74 @@ public class KakaoOAuthControllerTest {
     @MockBean
```

Comment:
MockBean이란 것은 실제로 Bean 대신 Mock을 주입한다는 뜻입니다. 그러나 KakaoTalkService에서는 RestTemplate을 빈으로 주입받아 사용하고 있지 않죠? 주입받는 거 대신 KakaoTalkService 객체가 싱글톤으로 생성될 때 같이 생성됩니다. 따라서 여기서 MockBean으로 만들어 바인딩한다해도 KakakoTalkService에서 사용하는 RestTemplate은 이 test에서 바인딩한 restTemplate이 아니게 됩니다.

이 문제를 해결하기 위해선 KakaoTalkService의 restTemplate을 `@Autowired`로 주입받아야 합니다. Configure 클래스를 만들어 RestTemplate을 Bean으로 만들어서 주입해보세요 :)

Reply:
말씀하신대로 수정해보니 KOE320 에러는 더 이상 발생하지 않습니다! 그렇지만 아직 getAccessToken 메서드에서 "org.springframework.http.ResponseEntity.getBody()" because "response" is null
에러가 발생합니다... 해당 에러 수정 후 2단계 다시 올리겠습니다!